### ISSUE TITLE
[Req] Way to know when players get down and when they die.

### ISSUE THREAD

**Issue by FoolHen** [VeniceUnleashed/VeniceUnleashed#464]
For RM we need to know when players get down and when they die completely. ``Player:Killed`` event doesn't work as its only called when a player dies and not when a player stops being revivable. 

Instead I could use ``ClientSoldier:HealthAction`` event that has the soldier and the new health action, but the issue with this one is that when the soldier isn't alive ``SoldierEntity.player`` is nil, so i can't know to what player each soldier belongs to. 
We either need ``SoldierEntity.player`` to return the correct player instead of nil if the soldier is dead or the player param in ``ClientSoldier:HealthAction`` event.

**Comment by FlashHit** [comment]
As in #226 listed <code>Player:Death(playerEntity) // This event occurs when a player is decisively dead, and cannot be revived; they will only return to the game by respawning.</code>
It would be nice to have that. 
I made a Revive Limiter Plugin and the problem is<code>admin.killPlayer</code> isn't working on dead players. Like I want them to stop being revivable after n kills. So I use the <code>admin.movePlayer</code> command on <code>Player:Killed</code> what isn't that elegant but still working on vanilla. I also added a System.Timer to move the player back when he stops being revivable. Moving him back straight without waiting would make the player revivable again. So this is the only way for me to do it. I also could use vars.playerManDownTime 0 but then the problem is that all other players wouldn't be able to receive a revive too while they are also dead.
So <code>Player:Death</code> event would be great to avoid the System.Timer. But in my case it would be even better if you add something like <code>admin.killPlayerNoRev</code> that makes the dead soldier stop being revivable. 
Otherwise I have the problem with full servers. Because I can't move players if the server is full.

In short: it would be great to have commands to make players down and to make them dead, and also to have events to know when players get down and when they die.

**Comment by FlashHit** [comment]
I think you can use PlayerManager and compare player.corpse with soldier.
```lua
Events:Subscribe('Soldier:HealthAction', function(soldier, action)
        if soldier.player == nil then  
                local playerdown = PlayerManager:GetLocalPlayer()
	        if playerdown.corpse == soldier then
		        print("playerdown.name")
		end
	else 
		 print("soldier.player.name")
	end
end)

**Comment by FoolHen** [comment]
Yeah that's what I do, looping through all players to which one died or whatever (server). It's better to have it as a parameter instead.

**Comment by OrfeasZ** [comment]
To provide the player as a parameter I'd have to loop through them in the same way you do, negatively affecting performance for anyone who's not interested in the player assigned to that soldier entity. You should handle this yourself the same way you are right now.

**Comment by FlashHit** [comment]
Is there no event where the server updates the deaths from the players? I mean the server knows if I died 1 time or 2 times. So it is updating the player deaths. And stop being revivable and getting 1 death added to total deaths is the same moment.
Maybe you could find another way to provide such an event. Idk. just a thought.

**Comment by FlashHit** [comment]
Implemented in `19412`.
